### PR TITLE
Fix - #13 Make the HomePage responsive for all the devices

### LINKED ISSUE #13: Make the HomePage responsive for all the devices
Presently the font looks too big and it doesn't have proper styling which is not suitable for the mobile devices
image

We have to make it responsive for devices 
1. Greater than desktop
2. Desktop devices
3. Tablet devices
4. mobile devices

### PR DESCRIPTION
**Pull request by Deblin798**
## Summary

Fixed Header Part for mobile screen and other CSS changes

## Description
CSS changes to fix Mobile and Tablet devices. Used Tailwind CSS to add mobile and tablet compatibility. Continued the changes for Issues #13 

Creating a issue for refactoring of all the md: to sm:

## Images

Screenshot 2024-01-17 003538
Screenshot 2024-01-17 003457

## Issue(s) Addressed

_Enter the issue number of the bug(s) that this PR fixes_

Closes #13 

## Prerequisites

- [x] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

**Comment by Deblin798** [comment]
@krishnaacharyaa apologies for the delay, I have addressed the PR comments. Please let me know if there's any other changes that needs to be taken care of!

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Hi @Deblin798, Thank you for the PR.
I guess you have missed these previous comments fixes in this PR, kindly check on that.
> Can we do the folowing updates in mobile devices

> Title of featured posts to be max of 1 line... (similar to All blogs post)
Title of Latest posts to be max of 1 line... (similar to All blogs post)

Kindly follow the contributing guidelines, you are missing the 2nd and 3rd one.
The PR just says "Merge Conflict" with 1 commit "merge conflict", not much that we can extract from it. 

Kindly address the above comments and we are good to merge this

**Comment by krishnaacharyaa** [comment]
@Deblin798, the skeleton for both cards looks out of sync to me, please look into that.
1. featured-post skeleton it should have only those many fields as the real post has. it currently has more than that
2. post-card it is not aligned in terms of height and width and also the no of components within the post card,

If you have any query regarding this, feel free to ask

**Comment by Deblin798** [comment]
Hi @krishnaacharyaa made the necessary changes and addressed the issues. Please have a look!

**Comment by krishnaacharyaa** [comment]
> Hi @krishnaacharyaa made the necessary changes and addressed the issues. Please have a look!

Thank you for the changes @Deblin798, Please follow contributing guidelines as it helps us keep track of the project history, please change the PR title according to the 3rd guideline https://github.com/krishnaacharyaa/wanderlust/blob/c3b11e57fa35786358d6c373911109fe03116bbf/.github/CONTRIBUTING.md

**Comment by Deblin798** [comment]
image
image
image
image

@krishnaacharyaa made changes to a few titles, descriptions, padding, text sizes, which include a few specific to mobile devices and a few to desktop site as well. Please have a look!

File: frontend/src/components/post-card.tsx [hunk 1]
```diff
@@ -8,21 +8,23 @@ export default function PostCard({ post }: { post: Post }) {
...
+          'md:mr-8 md:mt-4'
+        }`}
         onClick={() => navigate(`/details-page/${slug}/${post._id}`, { state: { post } })}
       >
         <img
           src={post.imageLink}
           alt={post.title}
-          className="h-48 w-full rounded-t-lg object-cover"
+          className="h-48 sm:h-64 w-full rounded-t-lg object-cover"
         />
```
Comment (krishnaacharyaa) [inline frontend/src/components/post-card.tsx:22]: I guess this should have been h-64 sm:h-48 right? Because the height is too big in the desktop screen

**Review by krishnaacharyaa — COMMENTED** [review]
LGTM , just one change and we'll merge this, 
I am witnessing this now, sorry missed it earlier...

And thank you for the regressive comments and addressing the reviews, really appreciate the efforts

File: frontend/src/components/post-card.tsx (hunk 1 above)
Comment (Deblin798) [reply frontend/src/components/post-card.tsx:22]: Sure thing let me do that!

**Review by krishnaacharyaa — APPROVED** [review]
Thank you, merging this now

**Pull request merged**